rfcs/010-live-assessment-moderation.md

```diff
@@ -0,0 +1,118 @@
+
+## Potential Per-Question Progress Dashboard
+
+Depending on how it would turn out, it may also be possible to create a secondary dashboard containing specialized per-question progress metrics. These metrics would be separate from the overall progress metric and might be capable of containing the following data, inheriting from existing assessment events or potentially just being a display of those events directly:
+
+- Timestamp when students click into or click out of a particular question
+- Timestamp when students have clicked save on a question (though this may be redundant)
+- Other metrics TBD based on further required analysis and demand
+
+This would allow a new dashboard containing row-by-row overviews relating to each question in an assessment. The setup would be similar to that of the students page, but the physical size of the proposed charts prohibit inclusion of the metric on the students page without it becoming visibly larger.
+
+Each row would correspond to a single student, and there would only be columns for student info and the timescale charts.
+
+These timescale charts are a timeline, bounded by the start and end of an assessment, along which events would be marked. On the y-axis of the chart would be the list of question numbers, in increasing order bottom to top or top to bottom.
+
+Question-save events could be marked as single markers, but events involving clicking into and out of a question can be represented as continuous bars from the click-in timestamp to the click-out timestamp. This allows the chart to show how long a student has attempted a question for and when.
+
+Two alternatives are possible. One option is to have this stacked per-question display where events for a particular question are in different rows of the chart. This would make it easier to visualize question-specific events.
+
+Alternatively, since time is linear, a singular row containing all the events and separate labels for question-specific events would fit in a smaller space, but at the same time it would be more difficult to view question-specific progress. The first option is recommended.
+
+Finally, an overall summary chart contiaining perhaps a heatmap-style version of the option 1 chart averaged over all students could be placed at the top of this dashboard. The heatmap-like color scaling would indicate how many students are inside a particular question at a particular moment in time, with redder or greyer colors for zero and bluer colors for higher concentrations of students. At any moment, it is possible to compute how many students are on a specific question by reading the question number of the most recent click-in event for that student.
+
+The idea is that by having this dashboard, anomalies such as questions immediately skipped or questions spent more time on than expected would then be trackable by professors and perhaps indicative of something wrong.
+

```

Comment:
Since students may not write their questions in the correct order, we may want to address how this dashboard would cover basic information about all questions. It should also consider how to handle question alternatives ("choose N of M").